ImageBlob fails to be retrieved from PostgreSQL database

##### **Overview of the issue**

While working #134 I tried to change the devDatabaseType to postgresql for the mvn-oauth-angular sample and the E2E tests failed.

Exception is:
```
Unexpected error occurred: org.hibernate.exception.DataException: could not execute query
javax.persistence.PersistenceException: org.hibernate.exception.DataException: could not execute query
...
Caused by: org.postgresql.util.PSQLException: Bad value for type long
```

##### **Reproduce the error**

See: https://github.com/jhipster/generator-jhipster-micronaut/pull/134/commits/24191263b63e9d1404de99ab3ae73ed1f869e2e6

The exception is visible when trying to load the SpaceEvent entity from the UI


-   [x] Checking this box is mandatory (this is just to show you read everything)

--- a/generators/constants.js
+++ b/generators/constants.js
@@ -18,11 +18,23 @@
  */
 
 module.exports = {
-    sqlDbDrivers: {
-        h2Disk: 'org.h2.Driver',
-        h2Memory: 'org.h2.Driver',
-        mysql: 'com.mysql.jdbc.Driver',
-        postgresql: 'org.postgresql.Driver',
+    sqlDb: {
+        h2Disk: {
+            driver: 'org.h2.Driver',
+            dialect: 'io.github.jhipster.domain.util.FixedH2Dialect',
+        },
+        h2Memory: {
+            driver: 'org.h2.Driver',
+            dialect: 'io.github.jhipster.domain.util.FixedH2Dialect',
+        },
+        mysql: {
+            driver: 'com.mysql.jdbc.Driver',
+            dialect: 'org.hibernate.dialect.MySQL5InnoDBDialect',
+        },
+        postgresql: {
+            driver: 'org.postgresql.Driver',
+            dialect: 'io.github.jhipster.domain.util.FixedPostgreSQL10Dialect',
+        },
     },
     versions: {
         micronaut: '2.0.1',

--- a/generators/server/templates/src/main/resources/application-dev.yml.ejs
+++ b/generators/server/templates/src/main/resources/application-dev.yml.ejs
@@ -10,10 +10,12 @@
 # https://www.jhipster.tech/common-application-properties/
 # ===================================================================
 <%_
-const { sqlDbDrivers } = MN_CONSTANTS;
+const { sqlDb } = MN_CONSTANTS;
 _%>
 <%_ if (databaseType === 'sql') { _%>
 <%_
+  const dbInfo = sqlDb[devDatabaseType];
+
   const dbUrlConfig = {
     databaseName: baseName,
     hostname: 'localhost'
@@ -42,7 +44,7 @@ datasources:
     url: "<%-dbUrl%>"
     username: "<%=dbUser%>"
     password: "<%=dbPass%>"
-    driverClassName: "<%=sqlDbDrivers[devDatabaseType]%>"
+    driverClassName: "<%=dbInfo.driver%>"
     poolName: Hikari
     autoCommit: false
 
@@ -55,29 +57,25 @@ liquibase:
 
 jpa:
   default:
-    packages-to-scan: <%=packageName%>.domain
+    packages-to-scan:
+      - <%=packageName%>.domain
+      - <%=packageName%>.repository
+    show-sql: false
     properties:
-      hibernate:
+      hibernate.dialect: "<%=dbInfo.dialect%>"
 <%_ if (['ehcache', 'caffeine', 'hazelcast', 'infinispan', 'memcached'].includes(cacheProvider)) { _%>
-        javax:
-          cache:
-            missing_cache_strategy: create
+      hibernate.javax.cache.missing_cache_strategy: create
 <%_ } _%>
-        hbm2ddl:
-          auto: none
-        show_sql: true
-        id:
-          new_generator_mappings: true
-        connection:
-          provider_disables_autocommit: true
-        cache:
+      hibernate.id.new_generator_mappings: true
+      hibernate.connection.provider_disables_autocommit: true
+      hibernate.connection.autocommit: false
 <%_ if (enableHibernateCache) { _%>
-          use_second_level_cache: true
+      hibernate.cache.use_second_level_cache: true
 <%_ } else { _%>
-          use_second_level_cache: false
+      hibernate.cache.use_second_level_cache: false
 <%_ } _%>
-          use_query_cache: false
-        generate_statistics: false
+      hibernate.cache.use_query_cache: false
+      hibernate.generate_statistics: false
 <%_ } _%>
 
 logging:

--- a/generators/server/templates/src/main/resources/application-prod.yml.ejs
+++ b/generators/server/templates/src/main/resources/application-prod.yml.ejs
@@ -10,7 +10,8 @@
 # https://www.jhipster.tech/common-application-properties/
 # ===================================================================
 <%_
-const { sqlDbDrivers } = MN_CONSTANTS;
+const { sqlDb } = MN_CONSTANTS;
+const dbInfo = sqlDb[prodDatabaseType];
 _%>
 logging:
   level:
@@ -47,10 +48,10 @@ datasources:
     url: ${JDBC_URL:`<%-dbUrl%>`}
     username: "<%=dbUser%>"
     password: "<%=dbPass%>"
-    driverClassName: "<%=sqlDbDrivers[prodDatabaseType]%>"
+    driverClassName: "<%=dbInfo.driver%>"
+    autoCommit: false
     hikari:
       poolName: Hikari
-      auto-commit: false
       maximum-pool-size: 10
       data-source-properties:
         cachePrepStmts: true
@@ -62,24 +63,24 @@ datasources:
 jpa:
   default:
     packages-to-scan:
-    - <%=packageName%>.domain
-    - <%=packageName%>.repository
-  database-platform: org.hibernate.dialect.MySQL5InnoDBDialect
-  database: MYSQL
-  show-sql: false
-  properties:
+      - <%=packageName%>.domain
+      - <%=packageName%>.repository
+    show-sql: false
+    properties:
+      hibernate.dialect: "<%=dbInfo.dialect%>"
 <%_ if (['ehcache', 'caffeine', 'hazelcast', 'infinispan', 'memcached', 'redis'].includes(cacheProvider)) { _%>
-    hibernate.javax.cache.missing_cache_strategy: create
+      hibernate.javax.cache.missing_cache_strategy: create
 <%_ } _%>
-    hibernate.id.new_generator_mappings: true
-    hibernate.connection.provider_disables_autocommit: true
+      hibernate.id.new_generator_mappings: true
+      hibernate.connection.provider_disables_autocommit: true
+      hibernate.connection.autocommit: false
 <%_ if (enableHibernateCache) { _%>
-    hibernate.cache.use_second_level_cache: true
+      hibernate.cache.use_second_level_cache: true
 <%_ } else { _%>
-    hibernate.cache.use_second_level_cache: false
+      hibernate.cache.use_second_level_cache: false
 <%_ } _%>
-    hibernate.cache.use_query_cache: false
-    hibernate.generate_statistics: false
+      hibernate.cache.use_query_cache: false
+      hibernate.generate_statistics: false
 
 # Replace by 'prod, faker' to add the faker context and have sample data loaded in production
 liquibase:

--- a/test-integration/samples/mvn-oauth-angular.jdl
+++ b/test-integration/samples/mvn-oauth-angular.jdl
@@ -12,7 +12,7 @@ application {
     clientThemeVariant primary
     creationTimestamp 1596598800266
     databaseType sql
-    devDatabaseType h2Disk
+    devDatabaseType postgresql
     dtoSuffix DTO
     embeddableLaunchScript false
     enableHibernateCache false

